cmd/openapi/transform.go

```diff
@@ -0,0 +1,121 @@
+package openapi
+
+import (
+	"context"
+	charm_internal "github.com/speakeasy-api/speakeasy/internal/charm"
+	"github.com/speakeasy-api/speakeasy/internal/model"
+	"github.com/speakeasy-api/speakeasy/internal/model/flag"
+	"github.com/speakeasy-api/speakeasy/internal/transform"
+	"github.com/speakeasy-api/speakeasy/internal/utils"
+	"os"
+)
+
+var transformCmd = &model.CommandGroup{
+	Usage:    "transform",
+	Short:    "Transform an OpenAPI spec using a well-defined function",
+	Commands: []model.Command{removeUnusedCmd, filterOperationsCmd, cleanupCmd},
+}
+
+type basicFlagsI struct {
+	Schema string `json:"schema"`
+	Out    string `json:"out"`
+}
+
+var basicFlags = []flag.Flag{
+	flag.StringFlag{
+		Name:                       "schema",
+		Shorthand:                  "s",
+		Description:                "the schema to transform",
+		Required:                   true,
+		AutocompleteFileExtensions: charm_internal.OpenAPIFileExtensions,
+	},
+	flag.StringFlag{
+		Name:        "out",
+		Shorthand:   "o",
+		Description: "write directly to a file instead of stdout",
+	},
+}
+
+var removeUnusedCmd = &model.ExecutableCommand[basicFlagsI]{
+	Usage: "remove-unused",
+	Short: "Given an OpenAPI file, remove all unused options",
+	Run:   runRemoveUnused,
+	Flags: basicFlags,
+}
+
+type filterOperationsFlags struct {
+	Schema       string   `json:"schema"`
+	Out          string   `json:"out"`
+	OperationIDs []string `json:"operations"`
+	Exclude      bool     `json:"exclude"`
+}
+
+var filterOperationsCmd = &model.ExecutableCommand[filterOperationsFlags]{
+	Usage: "filter-operations",
+	Short: "Given an OpenAPI file, filter down to just the given set of operations",
+	Run:   runFilterOperations,
+	Flags: append(basicFlags, []flag.Flag{
+		flag.StringSliceFlag{
+			Name:        "operations",
+			Description: "list of operation IDs to include (or exclude)",
+			Required:    true,
+		},
+		flag.BooleanFlag{
+			Name:         "exclude",
+			Shorthand:    "x",
+			Description:  "exclude the given operationIDs, rather than including them",
+			DefaultValue: false,
+		},
+	}...),
+}
+
+var cleanupCmd = &model.ExecutableCommand[basicFlagsI]{
+	Usage: "cleanup",
+	Short: "Cleanup the formatting of a given OpenAPI document",
```

Comment:
Maybe add a Long: description here explaining a bit more. Cleanup is a bit vague

Reply:
Good idea